llms/googleai/googleai.go

```diff
@@ -450,9 +450,10 @@ func convertTools(tools []llms.Tool) ([]*genai.Tool, error) {
 		}
 		genaiFuncDecl.Parameters = schema
 
-		genaiTools = append(genaiTools, &genai.Tool{
-			FunctionDeclarations: []*genai.FunctionDeclaration{genaiFuncDecl},
-		})
+		// google genai only support one tool, multiple tools must be embedded into function declarations:
+		// https://github.com/GoogleCloudPlatform/generative-ai/issues/636
+		// https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/function-calling#chat-samples
+		genaiTools[0].FunctionDeclarations = append(genaiTools[0].FunctionDeclarations, genaiFuncDecl)
```

Comment:
Perhaps I'm missing something obvious, but why do you need to predeclare `genatTools` and then do this `append`, can you not just create a single slice literal initialized to this?

Reply:
> Perhaps I'm missing something obvious, but why do you need to predeclare `genatTools` and then do this `append`, can you not just create a single slice literal initialized to this?

I just reused the `genaiTools` declaration before the change, now I initialized `genaiFuncDecls` and append all the funcs, `genaiTools` now initializes at the end.